src/servers/multi_server.cc

```diff
@@ -776,15 +785,109 @@ PrintModelStats(
       "deleting model stats message");
 }
 
+static volatile std::atomic<int> counter(0);
+static std::mutex mutex;
+std::condition_variable cv;
+
+void
+CreateAndRunTritonserverInstance(
+    std::vector<std::string> model_repository_paths, size_t thread_id,
+    bool verbose_level)
+{
+  std::unique_lock<std::mutex> lock(mutex);
+  counter++;
+  cv.wait(lock);
+
+  TRITONSERVER_ServerOptions* server_options = nullptr;
```

Comment:
Running this just one time means that it is very unlikely that the 2 threads will actually be doing anything at the same time. First it would be better to run with more threads / server instances. The models are small? So we should be able to run many instances /threads. 128? 256? And then the body of this function should iterate a large number of times (as many as we can and still have a ~5 minute runtime). That will allow the threads to interleave in many ways. Will need to Stop and Delete the server at the end of each iteration.

Reply:
Yes I had expanded the multi_server to be able to run for N threads. Will modify test.sh to run for 128 threads.

Reply:
Doesn't TRITONSERVER_ServerDelete call Stop anyway and would be called as soon as CreateAndRunTritonserverInstance ends?

Reply:
Could be, then you don't need stop. But you are going to create, run, delete in a look within CreateAndRunTrionServerInstance. You can't just run that once for each thread... that doesn't provide much of a test.

Reply:
Made the changes to run CreateAndRunTrionServerInstance 4 times and changed thread count to 64. The time to run the test is approximately 20 mins

Reply:
4 times is likely not enough, 100's would be better. Why does it take so long to run? Are we doing something slow during the iterations.

Reply:
We use locks during the creation of Tritonserver to avoid failures during cuda memory pool creation and enabling of metrics. As well as when destroying the cuda memory pool.  Additionally loading/unloading the models may be adding time. However the _major slowdown_ would be loading these models and _waiting_ for the server to be ready to perform inference. 

Just to confirm we want to run CreateAndRunTritonserverInstance 100 times in 64/128 threads?

Reply:
I'd still like to understand why we can only run this loop 32 * 10 = 320 times.

Reply:
Reduced the sleep I was using to 100ms. Now we can run 32 threads 32 times and it takes 20-22 mins. Reducing it further does not decrease runtime significantly. 
PS: Increasing to 2x threads (64) increases overall runtime by 2x as well.